### DISCUSSION TITLE
v1 Run Artisan Command via Invoke doesn't work

### DISCUSSION THREAD

**Discussion by etelford** [brefphp/bref#1479] [Support] [answered]
**Description:**

I am attempting to run an artisan command with the aws CLI `invoke` command. This technique is [described in the current docs](https://bref.sh/docs/runtimes/console.html#usage-without-serverless-framework), but I couldn't find an example with v1. I am not using `serverless.yml` with my installation.

**How to reproduce:**

When I run this on the command line:

```sh
aws lambda invoke --function-name MyFunction:alias--region us-west-2 --cli-binary-format raw-in-base64-out --payload '"artisan list"' response.json
```

In fact, no matter what I pass to `--payload`, I get the same error.

I get this as the response:

```json
{
    "errorType": "Exception",
    "errorMessage": "The command exited with a non-zero status code: 1",
    "stackTrace": [
        "#0 /var/task/vendor/bref/bref/src/Runtime/Invoker.php(34): {closure}()",
        "#1 /var/task/vendor/bref/bref/src/Runtime/LambdaRuntime.php(91): Bref\\Runtime\\Invoker->invoke()",
        "#2 /opt/bootstrap(65): Bref\\Runtime\\LambdaRuntime->processNextEvent()",
        "#3 {main}"
    ]
}
```

I have my function set up with both the `php-80-fpm` layer and the `php-80-console` layer and my entry point is `artisan`.

What am I not doing correctly?

**Comment by mnapoli** [comment, marked as answer]
Try with `--log-type=Tail` to see the logs of what went wrong.

**Reply by etelford** [reply]
That helped a ton! When I decoded the log, I see this:

```
In UnableToCreateDirectory.php line 18:
Unable to create a directory at /codebuild/output/src712652229/src/API/storage/app. 
```

My project directory is a bit unstandard, but from this I was able to see that `FilesystemServiceProvider::registerFlysystem()` was attempting to create the `local` disk in `/API/storage/app` instead of inside `/tmp`.

Thanks!

**Reply by mnapoli** [reply]
Glad to hear it has helped 🙌
